### ISSUE TITLE
[Question] Does SQLite RxStorage supports schema with the array-like fields

### ISSUE THREAD

**Issue by AndrewShapovalov** [pubkey/rxdb#4803]
Hi @pubkey! Excellent work you and the RxDB team have done in developing such a versatile database tool.

I am currently using RxDB in my project and have found it to be incredibly helpful with the LokiJS storage. However, I recently came across a potential challenge due to the limitations of SQLite storage, particularly when dealing with array types.

In my application, I have 2 schemas - _Inspection Type_ and _Issue Type_. The _Inspection Type_ schema contains a field called `issueTypeIds`, which serves as a reference to the _Issue Type_ schema. This implementation works flawlessly with LokiJS. However, as you know, SQLite does not support array types, which could pose an issue for my use case.
I would like to know if the same implementation with the [Population](https://rxdb.info/population.html) feature, which is documented to work similarly to population with mongoose, will be supported and fully functional with the SQLite storage. Specifically, I am concerned about the `issueTypeIds` field and its ability to maintain its reference integrity when using SQLite. In other words - does SQLite RxStorage supports schema with the array-like fields?

```
// Inspection type schema
{
  "version": 0,
  "primaryKey": "id",
  "type": "object",
  "properties": {
    "id": {
      "type": "string",
      "maxLength": 100
    },
    "name": {
      "type": "string"
    },
    "createdAt": {
      "type": "string"
    },
    "deletedAt": {
      "type": "string"
    },
    "updatedAt": {
      "type": "string"
    },
    "issueTypeIds": {
      "type": "array",
      "uniqueItems": true,
      "ref": "issuetype",
      "items": {
        "type": "string"
      }
    }
  },
  "required": ["id", "name", "createdAt", "updatedAt"]
}

// Issue type schema
{
  "version": 0,
  "primaryKey": "id",
  "type": "object",
  "properties": {
    "id": {
      "type": "string",
      "maxLength": 100
    },
    "name": {
      "type": "string"
    },
    "createdAt": {
      "type": "string"
    },
    "updatedAt": {
      "type": "string"
    },
    "deletedAt": {
      "type": "string"
    },
    "inspectionTypeIds": {
      "type": "array",
      "uniqueItems": true,
      "items": {
        "type": "string"
      }
    }
  },
  "required": ["id", "name", "createdAt", "updatedAt"]
}

**Comment by pubkey** [comment]
RxDB, like many other NoSQL databases, does not enforce strict relations.
Same goes if you use RxDB with any RxStorage, also SQLite. 
If you want loose relations, like with the RxDB-populate stuff, you array relation would likely work. You can try it out in any free RxStorage, like dexie.js

**Comment by sleoni** [comment]
Hi @pubkey, Thanks for the reply.
1. In the lokiJS RxStorage it works as expected, but we would like to try and see how it would behave on the SQLite implementation.
2. We know that the SQLite RxStorage is a paid library, therefore:
    a. Any chance you can let us try the library (in good faith)
    b. Or if you can't let us try it, can you test the schema above and let us know what the results look like
    
    The bottom line, we want to use the SQLite RxSorage and need to be able to query the 2 collections. For example the "Inspection type schema" and the "issueTypeIds" array inside of it.
